indexing-service/src/main/java/org/apache/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerProvisioningStrategy.java

```diff
@@ -482,10 +482,13 @@ private static ImmutableWorkerInfo workerWithTask(ImmutableWorkerInfo immutableW
     int parallelIndexTaskCapacity = task.getType().equals(ParallelIndexSupervisorTask.TYPE)
                                     ? task.getTaskResource().getRequiredCapacity()
                                     : 0;
+    int taskCapacity = task.getTaskResource().getRequiredCapacity();
+
     return new ImmutableWorkerInfo(
         immutableWorker.getWorker(),
         immutableWorker.getCurrCapacityUsed() + 1,
         immutableWorker.getCurrParallelIndexCapacityUsed() + parallelIndexTaskCapacity,
+        immutableWorker.incrementCurrCapacityUsedByTaskType(task.getType(), taskCapacity),
```

Comment:
Having this per worker is not what end user is expecting. Having a global limit/ratio is what the end user cares about.

Reply:
In effect, the overall ratio would be more or less the same as every worker would have this same ratio.
`parallelIndexTaskSlotRatio` also does the same thing, so it only makes sense to retain a similar logic.